xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -408,23 +415,41 @@ def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
 
         the_one_pass = PatternRewriteWalker(
             GreedyRewritePatternApplier([StencilConversion(return_targets, gpu=True)]),
-            apply_recursively=False,
+            apply_recursively=True,
             walk_reverse=True,
         )
         the_one_pass.rewrite_module(op)
+        type_pass = PatternRewriteWalker(
+            GreedyRewritePatternApplier(
+                [
+                    StencilTypeConversionFuncOp(),
```

Comment:
Will it work if we put this inside `StencilConversion`? 

Also, what is the intent behind separating these patterns from the rest? (assuming they can work with the `walk_reverse` flag) 
Can we just keep them inside the main bunch? (the main pass is now applied in a recursive fashion anyway)

Reply:
We can for the Func one, not as-is for the For one.

This would be best for sure, but I don't want to invest in that right now!
So I'll move the Func one to leav it where it was before, but leave the For on a second pass for now.
Happy to see a PR making it work in one pass if you want to look into it!